src/MinHeap.test.js

```diff
@@ -0,0 +1,116 @@
+var MinHeap = require('./MinHeap');
+
+describe('MinHeap', function() {
+  describe('#getFirst', function() {
+    describe('when initialized', function() {
+      it('first value is undefined', function() {
+        var heap = MinHeap();
+        expect(heap.getFirst()).toBeUndefined();
+      });
+    });
+
+    describe('after initialization', function() {
+      describe('when a value is added', function() {
+        it('first value is added value', function() {
+          var heap = MinHeap();
+          heap.add(1);
+          expect(heap.getFirst()).toBe(1);
+        });
+
+        describe('when the value is removed', function() {
+          it('first value is undefined', function() {
+            var heap = MinHeap();
+            heap.add(1);
+            heap.remove();
+            expect(heap.getFirst()).toBeUndefined();
+          });
+        });
+      });
+    });
+  });
+
+  describe('#add', function() {
+    describe('when empty', function() {
+      describe('add smaller value', function() {
+        it('smaller value is first value', function() {
+          var heap = MinHeap();
+          heap.add(1);
+          heap.add(0);
+          expect(heap.getFirst()).toBe(0);
+        });
+      });
+
+      describe('add larger value', function() {
+        it('smaller value is first value', function() {
+          var heap = MinHeap();
+          heap.add(1);
+          heap.add(2);
+          expect(heap.getFirst()).toBe(1);
+        });
+
+        describe('add smallest value', function() {
+          var heap = MinHeap();
+          heap.add(1);
+          heap.add(2);
+          heap.add(0);
+          expect(heap.getFirst()).toBe(0);
+        });
+      });
+    });
+  });
+
+  describe('#remove', function() {
+    describe('when empty', function() {
+      it('value is undefined', function() {
+        var heap = MinHeap();
+        expect(heap.remove()).toBeUndefined();
+      });
+    });
+
+    describe('when not empty', function() {
+      describe('when single value exists', function() {
+        it('is single value', function() {
+          var heap = MinHeap();
+          heap.add(1);
+          expect(heap.remove()).toBe(1);
+        });
+      });
+
+      describe('when two values exist', function() {
+        it('removes smaller value', function() {
+          var heap = MinHeap();
+          heap.add(1);
+          heap.add(2);
+          expect(heap.remove()).toBe(1);
+        });
+      });
+
+      describe('when ten values exist', function() {
```

Comment:
I would rather use `many` instead of a fixed number here. As you already noticed, keeping this in sync with what the code is doing is hard.

At the same time, if you take create the array of numbers to use for the tests just before this `describe`, you could make the title dynamic using the array length. But if you do that, remember to not mutate the array during the tests.